ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/SignedBeaconBlockUnblinder.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.datastructures.blocks;
+
+import java.util.function.Supplier;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.spec.datastructures.execution.ExecutionPayload;
+
+public interface SignedBeaconBlockUnblinder {
+
+  SignedBeaconBlockUnblinder executionPayload(
+      Supplier<SafeFuture<ExecutionPayload>> executionPayloadSupplier);
+
+  SignedBeaconBlock getSignedBlindedBeaconBlock();
+
+  SafeFuture<SignedBeaconBlock> unblind();
```

Comment:
This seems like quite an unintuitive set of methods.  It may be worth adding comments here to explain how they're expected to be used (and any required ordering).